### PR TITLE
Base command

### PR DESCRIPTION
**Pull request by akravch**
Create a base class for all commands.

#8

### CODE REVIEW

**Comment by akravch** [comment]
I don't think it is possible to really remove the `IBotCommand` though since `ICommandSelector` has a dependency on it.

File: SchedulerBot/SchedulerBot.Business.Commands/AddCommand.cs [hunk 1]
```diff
@@ -45,40 +42,32 @@ public class AddCommand : IBotCommand
...
 
-		#region IBotCommand Implementation
-
-		/// <inheritdoc />
-		public string Name { get; }
+		#region Overrides
 
 		/// <inheritdoc />
-		public async Task<CommandExecutionResult> ExecuteAsync(Activity activity, string arguments)
+		protected override async Task<CommandExecutionResult> ExecuteCoreAsync(Activity activity, string arguments)
```
Comment (pryabov) [inline SchedulerBot/SchedulerBot.Business.Commands/AddCommand.cs]: Possibly need to change naming to remove `ExecuteCoreAsync` -> `ExecuteCoreLogicAsync`?

File: SchedulerBot/SchedulerBot.Business.Commands/BotCommand.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		}
+
+		#endregion
+
+		#region Protected Properties
+
+		/// <summary>
+		/// Gets the logger.
+		/// </summary>
+		protected ILogger Logger { get; }
```
Comment (pryabov) [inline SchedulerBot/SchedulerBot.Business.Commands/BotCommand.cs]: protected properties should go after public properties but not after public methods

File: SchedulerBot/SchedulerBot.Business.Commands/AddCommand.cs (hunk 1 above)
Comment (akravch) [reply SchedulerBot/SchedulerBot.Business.Commands/AddCommand.cs]: Not sure about that, it is a pretty common pattern for naming an abstract method called from another one:

https://msdn.microsoft.com/en-us/library/system.web.mvc.controller.executecore(v=vs.118).aspx
https://msdn.microsoft.com/en-us/library/system.windows.uielement.measurecore(v=vs.110).aspx

So I'd leave it as it is.

File: SchedulerBot/SchedulerBot.Business.Commands/BotCommand.cs (hunk 2 above)
Comment (akravch) [reply SchedulerBot/SchedulerBot.Business.Commands/BotCommand.cs]: Fixed.

**Pull request merged**